concordium-consensus/src/Concordium/MultiVersion.hs

```diff
@@ -173,11 +184,11 @@ data Callbacks = Callbacks
       -- if an unrecognized update took effect.
       notifyRegenesis :: Maybe BlockHash -> IO (),
       -- |Notify a block was added to the tree. The arguments are
-      -- the hash of the block, and its absolute height.
-      notifyBlockArrived :: Maybe (BlockHash -> AbsoluteBlockHeight -> IO ()),
+      -- the hash of the block, its absolute height and whether this node was the baker.
+      notifyBlockArrived :: Maybe (BlockHash -> AbsoluteBlockHeight -> Bool -> IO ()),
       -- |Notify a block was finalized. The arguments are the hash of the block,
-      -- and its absolute height.
-      notifyBlockFinalized :: Maybe (BlockHash -> AbsoluteBlockHeight -> IO ())
+      -- its absolute height and whether this node was the baker.
```

Comment:
```suggestion
      -- its absolute height and whether the block was produced by the baker id configured for this node.
```

There is a difference between this sentence and what you wrote. Namely the same node can switch what baker id it is running with for example.